app/scripts/app.js

```diff
@@ -0,0 +1,253 @@
+              }),
+              $ocLazyLoad.load(
+              {
+                name:'ngTouch',
+                files:['bower_components/angular-touch/angular-touch.js']
+              })
+            }
+          }
+
+    })
+      .state('dashboard.chart',{
+        templateUrl:'views/chart.html',
+        url:'/chart?trackid',
+        controller:'ChartCtrl',
+        resolve: {
+          loadMyFile:function($ocLazyLoad) {
+            return $ocLazyLoad.load({
+              name:'chart.js',
+              files:[
+                'bower_components/angular-chart.js/dist/angular-chart.min.js',
+                'bower_components/angular-chart.js/dist/angular-chart.css'
+              ]
+            }),
+            $ocLazyLoad.load({
+                name:'sbAdminApp',
+                files:['scripts/controllers/chartContoller.js']
+            })
+          }
+        }
+    })
+      .state('dashboard.table',{
+        templateUrl:'views/table.html',
+        url:'/table'
+    })
+      .state('dashboard.panels-wells',{
+          templateUrl:'views/pages/panels-wells.html',
+          url:'/panels-wells'
+  /*        controller:'SinglePage',
+          resolve: {
+            loadMyFile:function($oclazyload) {
+              return $oclazyload.load({
+                name:'sbAdminApp',
+                files: ['scripts/controllers/SinglePageController.js']
+              })
+            }
+          }
+          */
+      })
+      .state('dashboard.buttons',{
+        templateUrl:'views/ui-elements/buttons.html',
+        url:'/buttons'
+    })
+      .state('dashboard.notifications',{
+        templateUrl:'views/ui-elements/notifications.html',
+        url:'/notifications'
+    })
+      .state('dashboard.typography',{
+       templateUrl:'views/ui-elements/typography.html',
+       url:'/typography'
+   })
+      .state('dashboard.icons',{
+       templateUrl:'views/ui-elements/icons.html',
+       url:'/icons'
+   })
+      .state('dashboard.error',{
+       templateUrl:(stateParams) => {
+       return 'views/error.html?path='+stateParams.path+'&status='+stateParams.status},
+       url:'/errorpage/:path/:status',
+       controller:'ErrorCtrl',
+       resolve: {
+         loadMyFile:function($ocLazyLoad) {
+           return $ocLazyLoad.load({
+               name:'sbAdminApp',
+               files:['scripts/controllers/ErrorController.js']
+           })
+         }
+       }
+   })
+  }])
+
+
+    .run(['$rootScope', '$location','$state','$cookieStore','$http',function ($rootScope,$location,$state,$cookieStore,$http)
+    {
+      console.log("came in")
+      $rootScope.globals = $cookieStore.get('globals') || {};
+      if($rootScope.globals.currentUser)
+      {
+        $http.defaults.headers.common = {'X-User': $rootScope.globals.currentUser.username, 'X-Token': $rootScope.globals.currentUser.authdata};
+      }
+      $rootScope.$on("$stateChangeStart", function(event, toState, toParams, fromState, fromParams){
```

Comment:
I a big fan of code style/conventions. It makes code a lot more readable :+1: so I would have formatted the `run` method like the following (note the whitespaces, braces, etc). I suggest to keep to this style, and also be consistent on indentation (there are some different indents within the files)!

```javascript
  .run(['$rootScope', '$state', '$cookieStore', '$http', function ($rootScope, $state, $cookieStore, $http) {
    console.log("came in")
    $rootScope.globals = $cookieStore.get('globals') || {};
    if ($rootScope.globals.currentUser) {
      $http.defaults.headers.common = {
        'X-User': $rootScope.globals.currentUser.username,
        'X-Token': $rootScope.globals.currentUser.authdata
      };
    }
    
    $rootScope.$on("$stateChangeStart", function(event, toState, toParams, fromState, fromParams) {
      console.log("Came to scopechange");
      if (toState.authenticate && !$rootScope.globals.currentUser) {
        console.log("checking here");
        $state.transitionTo("login");
        event.preventDefault();
      }
    });
  }])
```